### PR TITLE
HBP-184: Create an YML template file that generates a backup cronjob in OKD

### PR DESCRIPTION
**Pull request by rarescodemart**
- Have a separate YML file (than the one creating the TVB image initially) as we might want to edit this job
  - edit of the CronJob we found how to do only with the YML file, using the same name 
- We want to have a DB backup in EBRAINS OKD for the official TVB images
- maybe run it every weekend (to not propagate immediately a mistake) as the backup is not incremental

### CODE REVIEW

File: tvb_build/hbp_openshift_deployer/tvb_backup.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      name: backup
+      annotations:
+        volume.beta.kubernetes.io/storage-class: managed-nfs-storage
+        volume.beta.kubernetes.io/storage-provisioner: fuseim.pri/ifs
+    spec:
+      accessModes:
+        - "ReadWriteOnce"
+      resources:
+        requests:
+          storage: "2Ti"
```
Comment (liadomide) [inline tvb_build/hbp_openshift_deployer/tvb_backup.yml:39]: Why this big ? the DB original volume is only 100GB

File: tvb_build/hbp_openshift_deployer/tvb_deployer.yml [hunk 2]
```diff
@@ -292,6 +292,9 @@ objects:
                   cpu: '3'
                 requests:
                   cpu: '1'
+              env:
+                - name: APP_ENCRYPTION_HANDLER
+                  value: 'true'
```
Comment (liadomide) [inline tvb_build/hbp_openshift_deployer/tvb_deployer.yml:297]: @popaula937 We recently found this flag from missing from the YML by reading through the code (on what is expected). Are you aware of any reason why this should not be added here ?

**Review by liadomide — APPROVED** [review]

File: tvb_build/hbp_openshift_deployer/tvb_deployer.yml (hunk 2 above)
Comment (popaula937) [reply tvb_build/hbp_openshift_deployer/tvb_deployer.yml:297]: No reason against it, most probably was forgotten while working on this setup.

File: tvb_build/hbp_openshift_deployer/tvb_backup.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      jobTemplate:
+        spec:
+          template:
+            spec:
+              containers:
+                - name: tvb-postgres-backup
+                  image: docker-registry.ebrains.eu/okd_tools/postgresql_backups:1.1
+                  env:
+                    - name: DATABASE_HOST
+                      value: tvb-postgres
```
Comment (liadomide) [inline tvb_build/hbp_openshift_deployer/tvb_backup.yml:69]: Maybe document here that this should match in OKD the postgresql service name?

File: tvb_build/hbp_openshift_deployer/tvb_backup.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      suspend: false
+      successfulJobsHistoryLimit: 7
+      failedJobsHistoryLimit: 7
+      jobTemplate:
+        spec:
+          template:
+            spec:
+              containers:
+                - name: tvb-postgres-backup
+                  image: docker-registry.ebrains.eu/okd_tools/postgresql_backups:1.1
```
Comment (liadomide) [inline tvb_build/hbp_openshift_deployer/tvb_backup.yml:66]: Here we should change after we have a new release with ubuntu:latest and a newer pg_dump version

**Review by liadomide — CHANGES_REQUESTED** [review]

**Review by liadomide — APPROVED** [review]

**Pull request merged**